### PR TITLE
[MM-55480] Update logr to v2.0.21

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Update `logr` to ~~v2.0.20~~v2.0.21. That eliminates the need for some type conversion.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-55480

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/platform/log.go [hunk 1]
```diff
@@ -32,7 +31,10 @@ func (ps *PlatformService) initLogging() error {
 	// create the app logger if needed
 	if ps.logger == nil {
 		var err error
-		ps.logger, err = mlog.NewLogger(logr.MaxFieldLen(*ps.Config().LogSettings.MaxFieldSize))
+		ps.logger, err = mlog.NewLogger(
+			mlog.MaxFieldLen(*ps.Config().LogSettings.MaxFieldSize),
+			mlog.StackFilter("log"),
```
Comment (hanzei) [inline server/channels/app/platform/log.go:36]: Pigg backing this single change in here: It's useful to filter out the std library `log` package as it's more useful to see who actually triggered the log message.

File: server/public/shared/mlog/mlog.go [hunk 2]
```diff
@@ -98,32 +98,25 @@ func (lc LoggerConfiguration) toTargetCfg() map[string]logrcfg.TargetCfg {
...
-
-// Uint32 constructs a field containing a key and Uint32 value.
-var Uint32 = logr.Uint32
-
-// Uint constructs a field containing a key and Uint value.
-var Uint = logr.Uint
+// Int constructs a field containing a key and int value.
+func Int[T ~int | ~int8 | ~int16 | ~int32 | ~int64](key string, val T) Field {
+	return logr.Int[T](key, val)
+}
```
Comment (hanzei) [inline server/public/shared/mlog/mlog.go:104]: I don't think it's possible to alias a generic function.

Comment (wiggin77) [reply server/public/shared/mlog/mlog.go:104]: That's unfortunate.  My understanding is that inlining is also disabled for generic methods.

File: server/public/shared/mlog/mlog.go [hunk 3]
```diff
@@ -98,32 +98,25 @@ func (lc LoggerConfiguration) toTargetCfg() map[string]logrcfg.TargetCfg {
 // will be used to generate a string representation.
 var Any = logr.Any
 
-// Int64 constructs a field containing a key and Int64 value.
-var Int64 = logr.Int64
```
Comment (wiggin77) [inline server/public/shared/mlog/mlog.go:102]: This is a breaking change in a `public` package.  Any apps importing mlog will break.  I think we can deprecate the  specific typed methods but deleting them will require a full version bump.

Comment (hanzei) [reply server/public/shared/mlog/mlog.go:102]: `github.com/mattermost/mattermost-server/server/public` is not yet on v1. In https://docs.mattermost.com/upgrade/important-upgrade-notes.html, we stated 

> While this submodule has not yet shipped a v1 and will introduce breaking changes before stabilizing the API, it remains both forwards and backwards compatible with the Mattermost server itself.

One of the main points of the submodule was to allow breaking changes for now.

**Review by isacikgoz — APPROVED** [review]
LGTM, I wish there was a tool that can compute an impact analysis once we introduce a breaking change.

**Review by wiggin77 — APPROVED** [review]
LGTM

**Pull request merged**